From 36d81322f0bcc2ff92cd7c1bee9a0c175b988693 Mon Sep 17 00:00:00 2001
From: Jo Zzsi <jozzsicsataban@gmail.com>
Date: Sun, 11 May 2025 21:16:42 -0400
Subject: [PATCH 16/20] fix: improve hostonly sloppy mode

Resolve regression introduced by 8519dcd.

Fixes #1321 .

(cherry picked from commit bcf0093945b8d7b45006f410c52402450062a993)

Related: RHEL-97473
---
 modules.d/90kernel-modules/module-setup.sh | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/modules.d/90kernel-modules/module-setup.sh b/modules.d/90kernel-modules/module-setup.sh
index 72fbc41f..2daacb93 100755
--- a/modules.d/90kernel-modules/module-setup.sh
+++ b/modules.d/90kernel-modules/module-setup.sh
@@ -122,11 +122,11 @@ installkernel() {
 
         # if not on strict hostonly mode, install all known filesystems,
         # if the required list is not set via the filesystems variable
-        if [[ $hostonly_mode != "strict" ]]; then
-            if [[ -z $filesystems ]]; then
-                dracut_instmods -o -P ".*/(kernel/fs/nfs|kernel/fs/nfsd|kernel/fs/lockd)/.*" '=fs'
-            fi
-        elif [[ "${host_fs_types[*]}" ]]; then
+        if [[ $hostonly_mode != "strict" ]] && [[ -z $filesystems ]]; then
+            dracut_instmods -o -P ".*/(kernel/fs/nfs|kernel/fs/nfsd|kernel/fs/lockd)/.*" '=fs'
+        fi
+
+        if [[ $hostonly ]] && [[ "${host_fs_types[*]}" ]]; then
             hostonly='' instmods "${host_fs_types[@]}"
         fi
 
-- 
2.50.1

